packages/portfolio-deploy/src/portfolio-control.core.js

```diff
@@ -0,0 +1,109 @@
+/** @file core eval to give control of YMax contract to a smartWallet */
+
+import { makeTracer } from '@agoric/internal/src/debug.js';
+import { E } from '@endo/eventual-send';
+import { prepareContractControl } from './contract-control.js';
+import { orchPermit } from './orch.start.js';
+import {
+  deployPostalService,
+  permit as postalServicePermit,
+} from './postal-service.core.js';
+
+export { deployPostalService };
+
+/**
+ * @import {ChainStoragePresent} from './chain-info.core.js'
+ * @import {PortfolioBootPowers} from './portfolio-start.type.ts';
+ * @import {PostalServiceBoot} from './postal-service.core.js';
+ * @import {AttenuatedDepositPowers} from './attenuated-deposit.core.js';
+ */
+
+const contractName = 'ymax0';
+
+const trace = makeTracer('PCtrl');
+
+/**
+ * @param {BootstrapPowers &
+ *  ChainStoragePresent &
+ *  PortfolioBootPowers &
+ *  PostalServiceBoot &
+ *  AttenuatedDepositPowers
+ * } permitted
+ * @param {{ options: { ymaxControlAddress: string }}} config
+ */
+export const delegatePortfolioContract = async (permitted, config) => {
+  const { ymaxControlAddress } = config?.options || {};
+  assert(ymaxControlAddress);
+
+  await null;
+  const { consume, zone: bootZone } = permitted;
+
+  // we can only prepareContractControl() once per zone, so use a subzone
+  const ymaxZone = bootZone.subZone(contractName);
+  const makeContractControl = prepareContractControl(ymaxZone, {
```

Comment:
Are we accepting we will only ever delegate ymax once? What will we do if we need to revoke and replace the controller?

Reply:
This is now a heap zone.

Otherwise, we could change the subZone name.